File: frontend/src/shared/api/prescription.ts
import { apiClient } from './client';

export interface PrescriptionItemInput {
  name: string;
  dosage?: string | null;
  frequency?: string | null;
  duration?: string | null;
}

export interface PrescriptionTreatmentInput {
  name: string;
  instructions?: string | null;
}

export interface PrescriptionReminderInput {
  title: string;
  due_at?: string | null;
  note?: string | null;
}

export interface PrescriptionItem {
  id?: number;
  name: string;
  dosage: string | null;
  frequency: string | null;
  duration: string | null;
}

export interface PrescriptionTreatment {
  id?: number;
  name: string;
  instructions: string;
}

export interface PrescriptionReminder {
  id?: number;
  title: string;
  due_at: string | null;
  note: string | null;
}

export interface Prescription {
  id: number;
  consultation_id: number;
  pet_id: number;
  clinic_id: number;
  branch_id: number | null;
  veterinarian_id: number | null;
  diagnosis: string;
  treatment_notes: string;
  created_by: number | null;
  items: PrescriptionItem[];
  treatments: PrescriptionTreatment[];
  reminders: PrescriptionReminder[];
  created_at: string | null;
  updated_at: string | null;
}

export interface PrescriptionCreateData {
  consultation_id: number;
  pet_id: number;
  clinic_id?: number | null;
  veterinarian_id?: number | null;
  diagnosis: string;
  treatment_notes?: string | null;
  items?: PrescriptionItemInput[];
  treatments?: PrescriptionTreatmentInput[];
  reminders?: PrescriptionReminderInput[];
}

export interface PrescriptionPageMeta {
  page: number;
  page_size: number;
  size?: number;
  total: number;
  pages: number;
}

export interface PrescriptionListResponse {
  items: Prescription[];
  meta: PrescriptionPageMeta;
}

export interface PrescriptionListParams {
  page?: number;
  page_size?: number;
  pet_id?: number;
}

const BASE_PATH = '/prescriptions';

export async function createPrescription(
  data: PrescriptionCreateData,
): Promise<Prescription> {
  return apiClient.post<Prescription>(BASE_PATH, data);
}

export async function getPrescription(
  prescriptionId: number,
): Promise<Prescription> {
  return apiClient.get<Prescription>(`${BASE_PATH}/${prescriptionId}`);
}

export async function listPrescriptions(
  params: PrescriptionListParams = {},
): Promise<PrescriptionListResponse> {
  const queryParams = new URLSearchParams();
  queryParams.set('page', String(params.page ?? 1));
  queryParams.set('page_size', String(params.page_size ?? 20));

  if (params.pet_id) {
    queryParams.set('pet_id', String(params.pet_id));
  }

  const query = queryParams.toString();
  return apiClient.get<PrescriptionListResponse>(
    query ? `${BASE_PATH}?${query}` : BASE_PATH,
  );
}

File: backend/app/domain/entities/prescription.py
"""Entidades de dominio para prescripciones (BE-010)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PrescriptionItemInput(BaseModel):
    """Medicamento informativo (entrada)."""

    name: str = Field(..., min_length=1, max_length=200)
    dosage: str | None = Field(None, max_length=200)
    frequency: str | None = Field(None, max_length=200)
    duration: str | None = Field(None, max_length=200)


class PrescriptionTreatmentInput(BaseModel):
    """Tratamiento informativo (entrada)."""

    name: str = Field(..., min_length=1, max_length=300)
    instructions: str = Field(default="", max_length=2000)


class PrescriptionReminderInput(BaseModel):
    """Recordatorio interno (entrada)."""

    title: str = Field(..., min_length=1, max_length=300)
    due_at: datetime | None = None
    note: str | None = Field(None, max_length=1000)


class PrescriptionItem(BaseModel):
    """Medicamento informativo (salida, con ID)."""

    model_config = ConfigDict(from_attributes=True)

    id: int | None = None
    name: str = Field(..., min_length=1, max_length=200)
    dosage: str | None = None
    frequency: str | None = None
    duration: str | None = None


class PrescriptionTreatment(BaseModel):
    """Tratamiento informativo (salida, con ID)."""

    model_config = ConfigDict(from_attributes=True)

    id: int | None = None
    name: str = Field(..., min_length=1, max_length=300)
    instructions: str = Field(default="", max_length=2000)


class PrescriptionReminder(BaseModel):
    """Recordatorio interno (salida, con ID)."""

    model_config = ConfigDict(from_attributes=True)

    id: int | None = None
    title: str = Field(..., min_length=1, max_length=300)
    due_at: datetime | None = None
    note: str | None = None


class Prescription(BaseModel):
    """Receta veterinaria (salida completa, con items, tratamientos, recordatorios)."""

    model_config = ConfigDict(from_attributes=True)

    id: int | None = None
    consultation_id: int = Field(..., gt=0)
    pet_id: int = Field(..., gt=0)
    clinic_id: int = Field(..., gt=0)
    branch_id: int | None = None
    veterinarian_id: int | None = None
    diagnosis: str = Field(..., min_length=1, max_length=2000)
    treatment_notes: str = Field(default="", max_length=3000)
    created_by: int | None = None
    items: list[PrescriptionItem] = []
    treatments: list[PrescriptionTreatment] = []
    reminders: list[PrescriptionReminder] = []
    created_at: datetime | None = None
    updated_at: datetime | None = None


class PrescriptionCreate(BaseModel):
    """Datos de entrada para crear una receta (caso de uso)."""

    consultation_id: int = Field(..., gt=0)
    pet_id: int = Field(..., gt=0)
    clinic_id: int = Field(..., gt=0)
    branch_id: int | None = None
    veterinarian_id: int | None = None
    diagnosis: str = Field(..., min_length=1, max_length=2000)
    treatment_notes: str = Field(default="", max_length=3000)
    items: list[PrescriptionItemInput] = []
    treatments: list[PrescriptionTreatmentInput] = []
    reminders: list[PrescriptionReminderInput] = []
    created_by: int | None = None
